aws_quickstart/datadog_integration_api_call_v2.yaml

```diff
@@ -65,163 +65,193 @@ Resources:
+              api_url = event["ResourceProperties"]["ApiURL"]
+              account_id = event["ResourceProperties"]["AccountId"]
+              role_name = event["ResourceProperties"]["RoleName"]
+              aws_partition = event["ResourceProperties"]["AWSPartition"]
+              account_tags = event["ResourceProperties"]["AccountTags"]
+              cspm = event["ResourceProperties"]["CloudSecurityPostureManagement"]
+              metrics_disabled = event["ResourceProperties"]["DisableMetricCollection"]
 
               # Make the url Request
-              url = 'https://api.' + api_url + '/api/v1/integration/aws'
-              values = {
-                  'account_id': account_id,
-                  'role_name': role_name
-              }
-              if method == "POST":
-                  values["host_tags"] = host_tags
-                  values["aws_partition"] = aws_partition
-                  values["cspm_resource_collection_enabled"] = cspm == "true"
-                  values["metrics_collection_enabled"] = metrics_disabled == "false"
-
+              url = "https://api." + api_url + "/api/v2/integration/aws/accounts"
               headers = {
-                  'DD-API-KEY': api_key,
-                  'DD-APPLICATION-KEY': app_key,
-                  'Dd-Aws-Api-Call-Source': API_CALL_SOURCE_HEADER_VALUE,
+                  "DD-API-KEY": api_key,
+                  "DD-APPLICATION-KEY": app_key,
+                  "Dd-Aws-Api-Call-Source": API_CALL_SOURCE_HEADER_VALUE,
               }
-              data = json.dumps(values)
-              data = data.encode('utf-8')  # data should be bytes
-              request = Request(url, data=data, headers=headers)
-              request.add_header('Content-Type', 'application/json; charset=utf-8')
-              request.add_header('Content-Length', len(data))
-              request.get_method = lambda: method
 
-              # Send the url Request, store external_id
-              response = urllib.request.urlopen(request)
+              if method == "PATCH" or method == "DELETE":
+                  url = url + "/" + account_id
+
+              if method == "DELETE":
+                  # DELETE request has no body
+                  request = Request(url, headers=headers)
+              else:
+                  # Create the request body for POST and PATCH
+                  values = {
+                      "data": {
+                          "type": "account",
+                          "attributes": {
+                              "aws_account_id": account_id,
+                              "account_tags": account_tags,
+                              "aws_partition": aws_partition,
+                              "auth_config": {"role_name": role_name},
+                              "metrics_config": {
+                                  "enabled": (metrics_disabled == "false"),
+                                  "namespace_filters": {"include_all": True},
```

Comment:
We don't need to set this here - we can use the standard default if we don't provide this value.